instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/test/groovy/WithCurrentSpanInstrumentationTest.groovy

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import io.opentelemetry.instrumentation.test.AgentInstrumentationSpecification
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+import io.opentelemetry.test.annotation.SpanAttributesWithCurrentSpan
+
+import static io.opentelemetry.api.trace.SpanKind.INTERNAL
+
+/**
+ * This test verifies that auto instrumentation supports the
+ * {@link io.opentelemetry.instrumentation.annotations.WithCurrentSpan}
+ * and
+ * {@link io.opentelemetry.instrumentation.annotations.SpanAttribute}
+ * annotations.
+ */
+class WithCurrentSpanInstrumentationTest extends AgentInstrumentationSpecification {
+
+  def "should capture attributes in second span"() {
+    setup:
+    runWithSpan("external",
+        { new SpanAttributesWithCurrentSpan().withSpanAttributes("foo", "bar", null, "baz") })
+
+    expect:
+    assertTraces(1) {
+      trace(0, 2) {
+        span(0) {
+          name "external"
+          kind INTERNAL
```

Comment:
Seeing `external` and `INTERNAL` on the same span can be slightly misleading. Maybe rename `external` to `parent` or `root` for clarity?